cpp/src/traversal/mg/frontier_expand.cuh

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <graph.hpp>
+#include "frontier_expand_kernels.cuh"
+#include "vertex_binning.cuh"
+
+namespace cugraph {
+
+namespace mg {
+
+namespace detail {
+
+template <typename VT, typename ET, typename WT>
```

Comment:
Please consider lowercase and more descriptive typenames `<name>_t.` (the comment refers to `typename T`, not the non-type template parameter).